<launch>
    <arg name="resolution" default="0.25"/>

    <node pkg="octomap_server" type="octomap_server_node" name="octomap_server">
        <param name="resolution" value="$(arg resolution)" />
        <param name="frame_id" type="string" value="world" />
        <!--
        <param name="sensor_model/max_range" value="9.0" />
        <param name="sensor_model/min" value="0.01" />
        <param name="sensor_model/max" value="0.99" />
        <param name="sensor_model/hit" value="0.9" />
        <param name="sensor_model/miss" value="0.45" />
        <param name="color/r" value="0.1" />
        <param name="color/g" value="0.1" />
        <param name="color/b" value="0.1" />
        <param name="color/a" value="1.0" />
        <param name="occupancy_min_z" value="1.0" />
        <param name="height_map" value="false" />
        <param name="publish_free_space" value="false" />
        -->
        <remap from="cloud_in" to="uav1/camera/depth/points" />
        <!--
        <remap from="cloud_in1" to="/camera_1/depth/points" />
        <remap from="cloud_in2" to="/camera_2/depth/points" />
        <remap from="cloud_in3" to="/camera_3/depth/points" />
        -->
    </node>
<!--
    <group ns="uav1">
        <node pkg="octomap_server" type="octomap_server_node" name="octomap_server_1">
            <param name="resolution" value="$(arg resolution)" />
            <param name="frame_id" type="string" value="world" />
            <remap from="cloud_in" to="/camera_1/depth/points" />
        </node>
    </group>

    <group ns="uav2">
        <node pkg="octomap_server" type="octomap_server_node" name="octomap_server_2">
            <param name="resolution" value="$(arg resolution)" />
            <param name="frame_id" type="string" value="world" />
            <remap from="cloud_in" to="/camera_2/depth/points" />
        </node>
    </group>

    <group ns="uav3">
        <node pkg="octomap_server" type="octomap_server_node" name="octomap_server_3">
            <param name="resolution" value="$(arg resolution)" />
            <param name="frame_id" type="string" value="world" />
            <remap from="cloud_in" to="/camera_3/depth/points" />
        </node>
    </group>
-->
</launch>
